src/stream/src/executor/mview/materialize.rs

```diff
@@ -190,7 +208,19 @@ impl<S: StateStore, SD: ValueRowSerde> MaterializeExecutor<S, SD> {
                     }
                 }
                 Message::Barrier(b) => {
-                    self.state_table.commit(b.epoch).await?;
+                    let mutation = b.mutation.clone();
```

Comment:
Similar logic that checks self.conflict_behavior and downstream count exists many times, such as state table creation, write chunk and commit, and we have to carefully maintain the consistency between these states to ensure that the `commit_with_switch_consistent_op` is called only once when we switch from false to true.

I think the logic can be simplified to use the `is_consistent_op` flag of state table to the following
```
// On Write
if !state_table.is_consistent_op() || self.conflict_behavior == ConflictBehavior::NoCheck {
    // write without any check
} else {
    // write with check
}

// On commit
if state_table.is_consistent_op() && Self::new_downstream_created {
    state_table.commit_with_switch_consistent_op(true);
} else {
    state_table.commit();
}
```